rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -277,8 +293,19 @@ def query_gearbox_data_from_chain(
     pools_data = evm_inquirer.contracts.contract(
         CHAIN_ID_TO_DATA_COMPRESSOR[evm_inquirer.chain_id],
     ).call(node_inquirer=evm_inquirer, method_name='getPoolsV3List')
-    new_pools = []
+    new_pools: list[GearboxPoolData] = []
+    last_notified_ts = Timestamp(0)
+    chain_name = evm_inquirer.chain_id.to_name()
     for pool_data in pools_data:
+        last_notified_ts = maybe_notify_cache_query_status(
+            msg_aggregator=msg_aggregator,
+            last_notified_ts=last_notified_ts,
+            protocol=CPT_GEARBOX,
+            chain_name=chain_name,
+            processed=0,  # 0 because here we only query pools to find the total number of them, it will be incremented while processing the pools in `ensure_gearbox_tokens_existence`  # noqa: E501
+            total=len(new_pools),  # notifying the increase in len(pools) every 5 seconds as we query them  # noqa: E501
```

Comment:
Hmmm here this repeats again like in curve. And I guess you do it in other files too. Can't you create a version of `maybe_notify_cache_query_status` that's only for this? To find the total number of pools? It would just be calling this function with processed=0, and total=len(new_pools) in a for loop. And have a proper name so you don't need to repeat this comment and explain what it's for.

Something like (pseudocode):

```python
def find_total_number_of_pools(...) -> int:
for pool_data in pools_data:
    maybe_notify_cache_query_status(
            msg_aggregator=msg_aggregator,
            last_notified_ts=last_notified_ts,
            protocol=CPT_GEARBOX,
            chain_name=chain_name,
            processed=0,  # 0 because here we only query pools to find their total number. It will be incremented while processing the pools in `ensure_gearbox_tokens_existence`  # noqa: E501
total=len(new_pools), # notifying the increase in len(pools) every 5 seconds as we query them # noqa: E501
 )